### PR TITLE
feature/delete_temp_siks_when_election_ends

### PR DESCRIPTION
**Pull request by lucasmenendez**
Now the elections have a flag during creation (`tempSIKs`:boolean) which if set to true, the temporary SIKs (created using `RegisterSIKTx`) must be purged when the election ends.

### CODE REVIEW

**Review by lucasmenendez — COMMENTED** [review]
review done!

**Comment by p4u** [comment]
Finally... everything works as expected :)

@lucasmenendez please make a new review

**Comment by p4u** [comment]
should not be squashed, the commits are correctly ordered and with correct comments

File: tree/arbo/batch.go [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+		return nil, err
+	}
+	if nLeafs > t.thresholdNLeafs {
+		return t.addBatchInDisk(wTx, keys, values)
+	}
+	return t.addBatchInMemory(wTx, keys, values)
+}
+
+func (t *Tree) addBatchInDisk(wTx db.WriteTx, keys, values [][]byte) ([]Invalid, error) {
+	nCPU := flp2(runtime.NumCPU())
```
Comment (lucasmenendez) [inline tree/arbo/batch.go:61]: Why do you do this?

File: vochain/state/sik_test.go [hunk 2]
```diff
@@ -159,13 +159,14 @@ func TestAssignSIKToElectionAndPurge(t *testing.T) {
...
 	c.Assert(err, qt.IsNil)
+
 	// purge siks and check
 	c.Assert(s.PurgeSIKsByElection(pid), qt.IsNil)
 	c.Assert(err, qt.IsNil)
 	_, err = s.NoState(true).Get(key)
 	c.Assert(err, qt.IsNotNil)
-	_, err = s.SIKFromAddress(testAccount.Address())
-	c.Assert(err, qt.IsNil)
+	sik, err = s.SIKFromAddress(testAccount.Address())
```
Comment (lucasmenendez) [inline vochain/state/sik_test.go]: here `sik` var should be an underscore `_` because it is not used anymore.

File: tree/arbo/tree.go [hunk 3]
```diff
@@ -138,22 +139,27 @@ func NewTreeWithTx(wTx db.WriteTx, cfg Config) (*Tree, error) {
...
+	t := Tree{
+		db:              cfg.Database,
+		maxLevels:       cfg.MaxLevels,
+		thresholdNLeafs: cfg.ThresholdNLeafs,
+		hashFunction:    cfg.HashFunction,
+	}
+
 	t.emptyHash = make([]byte, t.hashFunction.Len()) // empty
+	var err error
+	t.emptyNode, _, err = t.newIntermediate(t.emptyHash, t.emptyHash)
```
Comment (lucasmenendez) [inline tree/arbo/tree.go:152]: Why the empty node is required?

File: tree/arbo/batch.go (hunk 1 above)
Comment (p4u) [reply tree/arbo/batch.go:61]: I did not change anything related the Add Batch.

File: tree/arbo/tree.go (hunk 3 above)
Comment (p4u) [reply tree/arbo/tree.go:152]: The empty node is used to compare in the up() function if both childs are empty. Precomputing and comparing is faster.

**Comment by p4u** [comment]
Is this ready @lucasmenendez? lets merge it!

**Comment by p4u** [comment]
I just created an issue with a couple of optimizations that can be done in the future: https://github.com/vocdoni/vocdoni-node/issues/1086

**Pull request merged**